### DISCUSSION TITLE
mail not send

### DISCUSSION THREAD

**Discussion by MrSuricate2** [symfony/symfony#52637] [Q&A]
Hello,

I have a problem, when I make the following command my server manages to send an email and I receive it, but when I configure the email sending from a symfony web page, I don't receive any email.

`php bin/console mailer:test my-email@example.com --from=contact@exemple.com`

```PHP
#[Route('/dashboard/sendEmail', name: 'app_dashboard_sendemail')]
    public function sendEmail(MailerInterface $mailer): Response
    {
        $email = (new Email())
            ->to('my-email@example.com')
            ->subject('Time for Symfony Mailer!')
            ->text('Sending emails is fun again!');
        try {
            $mailer->send($email);
        } catch (TransportExceptionInterface $e) {
            return new Response($e->getMessage());
        }
        return new Response("Email was sent");
    }
```
I followed the documentation and installed mailer and mime with composer.

Can you tell me what I did wrong or forgot?
    
Thanking you

**Comment by MatTheCat** [comment]
Hello,

You may have configured Messenger so that when you call `Mailer::send` a message is dispatched to a bus (see https://symfony.com/doc/current/mailer.html#sending-messages-async). If you don’t run the corresponding worker (see https://symfony.com/doc/current/messenger.html#consuming-messages-running-the-worker), the message won’t be handled and the mail won’t be sent.

The test command does not use Messenger: it sends the mail synchronously.

**Reply by ghost** [reply]
IIRC async mailing is enabled by default if messenger is installed.

**Comment by MrSuricate2** [comment]
> Hello,
> 
> You may have configured Messenger so that when you call `Mailer::send` a message is dispatched to a bus (see https://symfony.com/doc/current/mailer.html#sending-messages-async). If you don’t run the corresponding worker (see https://symfony.com/doc/current/messenger.html#consuming-messages-running-the-worker), the message won’t be handled and the mail won’t be sent.
> 
> The test command does not use Messenger: it sends the mail synchronously.

Thank you for your reply, but the e-mails still haven't been sent. On the other hand, they are stored in my database in the "messenger_messages" table, but the "delivered_at" column remains set to null. I therefore assume that the e-mails have not been sent.

**Reply by MatTheCat** [reply]
If `delivered_at` is `null`, the message has not been handled, thus the mail has not been sent.

**Reply by MrSuricate2** [reply]
I'm sorry, but I can't find this in the symfony documentation.

**Reply by MatTheCat** [reply]
If you cannot make use of the Messenger component despite its documentation I guess it would be best for you to remove it from your dependencies with `composer remove symfony/messenger`.

Note that emails would then be sent synchronously, which is generally not advised. Consider this solution as a workaround and try using Messenger later.

**Reply by MrSuricate2** [reply]
So even synchronously, the e-mails don't leave, I only receive them when I carry out the order via php but not from the web browser.

**Reply by ybico** [reply]
Any solution to the problem aforementioned by @MrSuricate2 ?
I'm having the same problem for over a month.

**Reply by MatTheCat** [reply]
Please provide a [reproducer](https://symfony.com/doc/current/contributing/code/reproducer.html).
